### PR TITLE
[CodeCompletion] 'case' keyword completion at the top of 'switch' stmt

### PR DESCRIPTION
**Pull request by rintaro**
rdar://problem/35943849

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/IDE/complete_keywords.swift [hunk 1]
```diff
@@ -401,3 +404,16 @@ func inExpr5() {
...
+func inSwitch(val: Int) {
+  switch val {
+  #^SWITCH_TOP^#
+  case 1:
+    foo()
+  #^SWITCH_IN_CASE^#
+  }
+// Begin completions
+// KW_CASE-DAG: Keyword[case]/None:                 case; name=case
+// KW_CASE-DAG: Keyword[default]/None:              default; name=default
```
Comment (jrose-apple) [inline test/IDE/complete_keywords.swift]: Bonus points (starter bug?): refuse to private `default` unless the completion's happening at the end of the switch.

Comment (rintaro) [reply test/IDE/complete_keywords.swift]: We should implement syntactic context aware keyword completion mechanism. Currently we provide `fallthrough`, `break`, et al. even in top-level.

**Comment by rintaro** [comment]
@swift-ci Please smoke test Linux platform.

**Pull request merged**